eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -282,24 +277,49 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
     }
   }
 
-  private def onSuccess(replyTo: ActorRef, result: PaymentSent): Unit = {
-    if (cfg.storeInDb) paymentsDb.updateOutgoingPayment(result)
-    replyTo ! result
-    if (cfg.publishEvent) context.system.eventStream.publish(result)
-    Metrics.SentPaymentDuration
-      .withTag(Tags.MultiPart, if (cfg.id != cfg.parentId) Tags.MultiPartType.Child else Tags.MultiPartType.Disabled)
-      .withTag(Tags.Success, value = true)
-      .record(System.currentTimeMillis - start, TimeUnit.MILLISECONDS)
-  }
-
-  private def onFailure(replyTo: ActorRef, result: PaymentFailed): Unit = {
-    if (cfg.storeInDb) paymentsDb.updateOutgoingPayment(result)
-    replyTo ! result
+  def myStop(request: SendPayment, result: PaymentEvent): State = {
```

Comment:
nit: similarly to `MultiPartPaymentLifecycle`, you could here take a `result: Either[PaymentFailed, PaymentSent]` to avoid the `INTERNAL_ERROR` case and the `case _ => ()`

Reply:
Done